src/Avalonia.Themes.Default/TextBox.xaml

```diff
@@ -38,12 +38,13 @@
                              Text="{TemplateBinding Watermark}"
                              IsVisible="{TemplateBinding Path=Text, Converter={x:Static StringConverters.NullOrEmpty}}"/>
                   <TextPresenter Name="PART_TextPresenter"
-                                 Text="{TemplateBinding DisplayText, Mode=TwoWay}"
+                                 Text="{TemplateBinding Text, Mode=TwoWay}"
```

Comment:
May be OneWay? Otherwise TextPresenter override TextProperty in TextBox like '******'
```
<TextBox Width="200" Name="TextBox"
         Watermark="Password Box"
         UseFloatingWatermark="True"
         PasswordChar="*"
         Text="Password" />
<TextBox Text="{Binding ElementName=TextBox, Path=Text}" />
```